pkg/geo/geomfn/translate_test.go

```diff
@@ -0,0 +1,122 @@
+// Copyright 2020 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package geomfn
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/cockroachdb/cockroach/pkg/geo"
+	"github.com/stretchr/testify/require"
+	"github.com/twpayne/go-geom"
+)
+
+func TestTranslate(t *testing.T) {
+	testCases := []struct {
+		desc   string
+		g      geom.T
+		deltas []float64
+	}{
+		{
+			desc:   "move a 2D point",
+			g:      geom.NewPointFlat(geom.XY, []float64{10, 10}),
+			deltas: []float64{5, -5},
+		},
+		{
+			desc:   "move a line string",
+			g:      geom.NewLineStringFlat(geom.XY, []float64{10, 15, 20, 30}),
+			deltas: []float64{1, 5},
+		},
+		{
+			desc:   "move a polygon",
+			g:      geom.NewPolygonFlat(geom.XY, []float64{0, 0, 5, 5, 0, 10, 0, 0}, []int{8}),
+			deltas: []float64{1, 1},
+		},
+		{
+			desc:   "move multiple 2D points",
+			g:      geom.NewMultiPointFlat(geom.XY, []float64{5, 10, -30, 40}),
+			deltas: []float64{1, 0.5},
+		},
+		{
+			desc:   "move multiple line strings",
+			g:      geom.NewMultiLineStringFlat(geom.XY, []float64{1, 1, 2, 2, 3, 3, 4, 4}, []int{4, 8}),
+			deltas: []float64{.1, .1},
```

Comment:
I'm getting unexpected behavior from converting an empty `LineString` to a `Geometry` and back to `geom.T`.

The code below will fail on the last `require.Nil` check for an empty `LineString` with error `Expected nil, but got: []float64{}`

Whereas the test will pass for an empty `Polygon`.

```go
	testCases := []struct {
		desc   string
		g      geom.T
		deltas []float64
	}{
		{
			desc:   "translate empty line string",
			g:      geom.NewLineString(geom.XY),
			deltas: []float64{1, 1},
		},
		{
			desc:   "translate empty polygon",
			g:      geom.NewPolygon(geom.XY),
			deltas: []float64{1, 1},
		},
	}

	for _, tc := range testCases {
		require.Nilf(t, tc.g.FlatCoords(), tc.desc)
		geometry, err := geo.NewGeometryFromGeomT(tc.g)
		require.NoError(t, err)

		tc.g, err = geometry.AsGeomT()
		require.NoError(t, err)
		require.Nilf(t, tc.g.FlatCoords(), tc.desc)
      }
```

Reply:
ideally we don't check `Nilf`, we should instead check whether the Len is 0.

also, you should run tests like this to avoid needing the `Nilf`:

```
t.Run(tc.desc, func(t *testing.T) {
   ... tests
})
```
that way you know the subtest that fails.

Reply:
Ah, makes sense, thanks!